metal/lock.h

```diff
@@ -4,11 +4,16 @@
 #ifndef METAL__LOCK_H
 #define METAL__LOCK_H
 
+#include <metal/memory.h>
+#include <metal/compiler.h>
+
 /*!
  * @file lock.h
  * @brief An API for creating and using a software lock/mutex
  */
 
+#define _METAL_STORE_AMO_ACCESS_FAULT 7
```

Comment:
This is already define in metal_exception_code_e;
May be metal_exception_code_e; should be brought out to interrupt.h

Reply:
`metal_exception_code_e` is defined in `metal/drivers/riscv,cpu.h`. Should I include that file?